From c4cec3024ee04402d5d90c8944102ba036e6cf80 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Wed, 15 May 2019 19:51:52 +0700
Subject: [PATCH 120/248] arm64: dts: r8a7796: Add INTC-EX device node

Add a device node for the Interrupt Controller for External Devices
(INTC-EX) on RZ/G2M, which serves external IRQ pins IRQ[0-5].

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a1.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a1.dtsi b/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
index e4c31aa..78b88f8 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
@@ -224,6 +224,22 @@
 			resets = <&cpg 408>;
 		};
 
+		intc_ex: interrupt-controller@e61c0000 {
+			compatible = "renesas,intc-ex-r8a774a1", "renesas,irqc";
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0 0xe61c0000 0 0x200>;
+			interrupts =	<GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 407>;
+			power-domains = <&sysc R8A774A1_PD_ALWAYS_ON>;
+			resets = <&cpg 407>;
+		};
+
 		prr: chipid@fff00044 {
 			compatible = "renesas,prr";
 			reg = <0 0xfff00044 0 4>;
-- 
2.7.4

